docs/user/reconciliation.md

```diff
@@ -0,0 +1,166 @@
+Because the number is bumped, the graph blocks until the previous `0000_03_*` are all complete before beginning the `0000_05_*` block.
+
+We are more relaxed for the initial install, because there is not yet any user data in the cluster to be worried about.
+So the graph nodes are all parallelized with the by-number ordering flattened out:
+
+<div style="text-align:center">
+  <img src="tasks-flatten-by-number-and-component.svg" width="100%" />
+</div>
+
+For the usual reconciliation loop (neither an upgrade between releases nor a fresh install), the flattened graph is also randomly permuted to avoid hanging on ordering bugs.
+
+## Synchronizing the graph
+
+The cluster-version operator spawns worker goroutines that walk the graph, pushing manifests in their queue.
+For each manifest in the node, the worker synchronizes the cluster with the manifest using a resource builder.
+On error (or timeout), the worker abandons the manifest, graph node, and any dependencies of that graph node.
+On success, the worker proceeds to the next manifest in the graph node.
+
+## Resource builders
+
+Resource builders synchronize the cluster with a manifest from the release image.
+The general approach is to generates a merged manifest combining critical spec properties from the release-image manifest with data from a preexisting in-cluster object, if any.
+If the merged manifest differs from the in-cluster object, the merged manifest is pushed back into the cluster.
+
+Some types have additional logic, as described in the following subsections.
+Note that this logic only applies to manifests included in the release image itself.
+For example, only [ClusterOperator](../dev/clusteroperator.md) from the release image will have the blocking logic described [below](#clusteroperator); if an admin or secondary operator pushed a ClusterOperator object, it would not impact the cluster-version operator's graph synchronization.
+
+### ClusterOperator
+
+The cluster-version operator does not push [ClusterOperator](../dev/clusteroperator.md) into the cluster.
+Instead, the operators create ClusterOperator themselves.
+The ClusterOperator builder only monitors the in-cluster object and blocks until it is:
+
+* Available
+* Either not progressing or, when the release image manifest has `status.versions` entries, listing at least the versions given in that manifest.
+    For example, an OpenShift API server ClusterOperator entry in the release image like:
+
+    ```yaml
+    apiVersion: config.openshift.io/v1
+    kind: ClusterOperator
+    metadata:
+      name: openshift-apiserver
+    spec: {}
+    status:
+      versions:
+        - name: operator
+          version: "4.1.0"
+    ```
+
+    would block until the in-cluster ClusterOperator reported `operator` at version 4.1.0.
+
+    The progressing check is deprecated and will be removed once all operators are reporting versions.
+* Not degraded (except during initialization, where we ignore the degraded status)
```

Comment:
there is already some content here https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusteroperator.md#conditions-and-installupgrade

Reply:
> there is already some content here...

Filed #274 to consolidate.